neural_modelling/src/neuron/plasticity/stdp/weight_dependence/weight_additive_two_term_impl.c

```diff
@@ -36,32 +47,32 @@ address_t weight_initialise(
 
     // Copy plasticity region data from address
     // **NOTE** this seems somewhat safer than relying on sizeof
-    int32_t *plasticity_word = (int32_t *) address;
-    plasticity_weight_region_data =
-            spin1_malloc(sizeof(plasticity_weight_region_data_t) * n_synapse_types);
-    if (plasticity_weight_region_data == NULL) {
+    additive_two_term_config_t *config = (additive_two_term_config_t *) address;
+
+    struct plasticity_weight_region_data_two_term_t *dtcm_copy =
+            plasticity_weight_region_data = spin1_malloc(
+                    sizeof(struct plasticity_weight_region_data_two_term_t) *
+                    n_synapse_types);
+    if (dtcm_copy == NULL) {
         log_error("Could not initialise weight region data");
         return NULL;
     }
-    for (uint32_t s = 0; s < n_synapse_types; s++) {
-        plasticity_weight_region_data[s].min_weight = *plasticity_word++;
-        plasticity_weight_region_data[s].max_weight = *plasticity_word++;
-        plasticity_weight_region_data[s].a2_plus = *plasticity_word++;
-        plasticity_weight_region_data[s].a2_minus = *plasticity_word++;
-        plasticity_weight_region_data[s].a3_plus = *plasticity_word++;
-        plasticity_weight_region_data[s].a3_minus = *plasticity_word++;
+    for (uint32_t s = 0; s < n_synapse_types; s++, config++) {
+        dtcm_copy[s].min_weight = config->min_weight;
+        dtcm_copy[s].max_weight = config->max_weight;
+        dtcm_copy[s].a2_plus = config->a2_plus;
+        dtcm_copy[s].a2_minus = config->a2_minus;
+        dtcm_copy[s].a3_plus = config->a3_plus;
+        dtcm_copy[s].a3_minus = config->a3_minus;
 
         log_debug("\tSynapse type %u: Min weight:%d, Max weight:%d, A2+:%d, A2-:%d,"
                 " A3+:%d, A3-:%d",
-                s, plasticity_weight_region_data[s].min_weight,
-                plasticity_weight_region_data[s].max_weight,
-                plasticity_weight_region_data[s].a2_plus,
-                plasticity_weight_region_data[s].a2_minus,
-                plasticity_weight_region_data[s].a3_plus,
-                plasticity_weight_region_data[s].a3_minus);
+                s, dtcm_copy[s].min_weight, dtcm_copy[s].max_weight,
+                dtcm_copy[s].a2_plus, dtcm_copy[s].a2_minus,
+                dtcm_copy[s].a3_plus, dtcm_copy[s].a3_minus);
     }
     log_debug("weight_initialise: completed successfully");
 
     // Return end address of region
-    return (address_t) plasticity_word;
+    return (address_t) config;
```

Comment:
Marking to again highlight a potential master issue.

Reply:
Ignore my misreading

Reply:
Both `master` and my code are correct. It's the `config` pointer that moves.